src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -93,18 +94,70 @@ def export_group_as_template(
 
 def deploy_arm_template(
         resource_group_name, template_file=None, template_uri=None, deployment_name=None,
-        parameters=None, mode='incremental', no_wait=False):
+        parameters=None, mode='incremental', no_wait=False, prompt_for_parameters=False):
     return _deploy_arm_template_core(resource_group_name, template_file, template_uri,
-                                     deployment_name, parameters, mode, no_wait=no_wait)
+                                     deployment_name, parameters, mode, no_wait=no_wait,
+                                     prompt_for_parameters=prompt_for_parameters)
 
 def validate_arm_template(resource_group_name, template_file=None, template_uri=None,
-                          parameters=None, mode='incremental'):
+                          parameters=None, mode='incremental', prompt_for_parameters=False):
```

Comment:
It seems to me we tend to opt-out of interactivity rather than opt-in, which would make this a --suppress-prompt flag instead of the opposite. One could argue that this flag shouldn't be needed at all.  If you are in interactive mode, you probably don't want to type --prompt to get the prompting behavior, and if you are running a script, it should just detect the lack of TTY and fail if the parameters aren't provided. In that case, when would you ever need to use --suppress-prompt (if it existed)?